lib/msf/base/sessions/command_shell.rb

```diff
@@ -421,52 +421,68 @@ def cmd_download(*args)
 
     # Check if src exists
     if !_file_transfer.file_exist?(src)
-      print_error("The target file does not exist")
+      print_error('The target file does not exist')
       return
     end
 
+    if (::File.basename(dst) != ::File.basename(src))
+      # The destination when downloading is a local file so use this system's separator
+      dst += ::File::SEPARATOR + ::File.basename(src)
+    end
+    dst_dir = ::File.dirname(dst)
+    ::FileUtils.mkdir_p(dst_dir) if dst_dir and not ::File.directory?(dst_dir)
+
     # Get file content
-    print_status("Download #{src} => #{dst}")
+    # match the output style of the Meterpreter equivalent
+    print_status("Downloading: #{src} -> #{dst}")
     content = _file_transfer.read_file(src)
 
     # Write file to local machine
     File.binwrite(dst, content)
-    print_good("Done")
+    print_status("Completed  : #{src} -> #{dst}")
   end
 
   def cmd_upload_help
     print_line("Usage: upload [src] [dst]")
     print_line
     print_line("Uploads load file to the victim machine.")
-    print_line("This command does not support to upload a FOLDER yet")
+    print_line("This command does not support directories")
     print_line
   end
 
   def cmd_upload(*args)
     if args.length != 2
-      # no argumnets, just print help message
+      # no arguments, just print help message
       return cmd_upload_help
     end
 
     src = args[0]
     dst = args[1]
 
+    if dst.blank?
+      dst = ::File.basename(src)
+    elsif _file_transfer.directory?(dst)
+      fs_sep = platform == 'windows' ? '\\' : '/'
+      dst += fs_sep unless dst.end_with?(fs_sep)
+      dst += ::File.basename(src)
+    end
+
     # Check target file exists on the target machine
     if _file_transfer.file_exist?(dst)
-      print_warning("The file <#{dst}> already exists on the target machine")
-      unless prompt_yesno("Overwrite the target file <#{dst}>?")
+      print_warning('The target file already exists')
+      unless prompt_yesno("Overwrite the target file #{dst}?")
         return
       end
     end
 
+    print_status("Uploading  : #{src} -> #{dst}")
     begin
       content = File.binread(src)
```

Comment:
Should this not be `::` given we are dealing with the source file? Would match the pattern above.